File: Python/LinkedList/Kth from End of Linked List.py
//https://www.geeksforgeeks.org/problems/nth-node-from-end-of-linked-list/1
class Solution:
    def getKthFromLast(self, head, k):
        #code here
        total = 0
        temp = head
        while temp:
            total += 1
            temp = temp.next
        
        # If k is greater than the number of nodes, return -1
        if k > total:
            return -1
        
        # Find the (total - k)th node
        x = total - k
        temp = head
        while x > 0:
            temp = temp.next
            x -= 1
        
        return temp.data

File: Java/Linked List/Kth from End of Linked List.java
//https://www.geeksforgeeks.org/problems/nth-node-from-end-of-linked-list/1
class Solution {

    // Function to find the data of kth node from
    // the end of a linked list.
    int getKthFromLast(Node head, int k) {
        // Your code here
        int total = 0;
        Node temp = head;

        // Step 1: Count total nodes
        while (temp != null) {
            total++;
            temp = temp.next;
        }

        // Step 2: If k is larger than total nodes
        if (k > total) return -1;

        // Step 3: Find (total - k)th node from the start
        int x = total - k;
        temp = head;
        while (x > 0) {
            temp = temp.next;
            x--;
        }

        return temp.data;
    }
}
